packages/signing-utils/src/ssh-client.spec.ts

```diff
@@ -0,0 +1,182 @@
+import sinon from 'sinon';
+import { SSHClient } from './ssh-client';
+import { expect } from 'chai';
+
+describe('SSHClient', function () {
+  let sshClient: SSHClient;
+  let sandbox: sinon.SinonSandbox;
+
+  beforeEach(function () {
+    const sshClientOptions = {
+      host: 'example.com',
+      port: 22,
+      username: 'admin',
+    };
+    sshClient = new SSHClient(sshClientOptions);
+    sandbox = sinon.createSandbox();
+  });
+
+  afterEach(function () {
+    sandbox.restore();
+  });
+
+  it('connects successfully on ready', async function () {
+    const connectStub = sandbox
+      .stub(sshClient['sshConnection'], 'connect')
+      .resolves(sshClient['sshConnection']);
+    await Promise.all([
+      sshClient.connect(),
+      sshClient['sshConnection'].emit('ready'),
+    ]);
```

Comment:
Does `.emit('ready')` actually return a Promise? I can't find anything to imply that it does

If the goal here is to execute code while the Promise returned from `.connect()` is pending, I'd just do that in the explicit (and shorter) version, i.e.

```suggestion
    const connectPromise = sshClient.connect();
    sshClient['sshConnection'].emit('ready');
    await connectPromise;
```

(ditto below)

Reply:
Yeah, good catch. I didn't look that closely at this test file, I ended up re-working it.